paper/paper.md

```diff
@@ -0,0 +1,105 @@
+
+mBuild functions as a molecular builder, meticulously crafted with extensive utilities for creating, loading, and manipulating the positions of atoms and molecules, along with managing their connectivity through bonds[@klein2016hierarchical]. These utilities have been applied in various projects to explore a wide range of structures and applications, as well as integrated into other scientific libraries[@quach2022high; @albooyeh2023flowermd; @ma2022dynamics].
+
+Foyer assumes the role of atom typing for the created structures, involving the identification and assignment of interaction parameters to each atom or group of atoms[@klein2019formalizing]. This process entails matching the connectivity (bond graph) of the provided structure with the SMARTS grammar of the corresponding atom type, defining the interactional parameters[@klein2019formalizing]. The use of a graph matching method, departing from the traditional approach of matching via atom indices, allows for a more flexible parameterization. This feature proves particularly advantageous in the study of functionalized polymers, whose structures consistently deviate slightly from the standard polymer[@quach2022high].
+
+The parameterization step introduces additional information, requiring a more intricate data structure for representation. Beyond the initial details concerning positions and connectivity established during system construction, the new structure incorporates supplementary metadata and interaction parameters. These encompass not only atoms and bonds but also extend to angles, dihedrals, and improper dihedrals, along with specific interaction parameters associated with each of these objects. This necessitates the development of data structures that are capable of:
+- Supporting a variety of models
+- Providing flexibility for exotic potentials
+- Being compatible with existing community tools
+- Being extensible (to support new simulation models/engines/workflows)
+
+Currently, there are data structures designed to represent these systems, such as ParmEd and OpenMM[@shirts2016lessons; @eastmann2017openmm]. However, these data structures are tailored to a specific subset of simulation workflows/ecosystems, sacrificing some generality. This limitation includes, but is not limited to, the hard-coding and assumption of potential (interactional) expressions and units. They lack the generality that MoSDeF seeks, such as the ability to define and store arbitrary potential expressions or unit systems. Integrating these new features would require a major overhaul of these data structures, potentially impacting existing simulation workflows and may not be appealing to current stakeholders in those projects. Hence, we have developed a new data structure called the General Molecular Simulation Object, or GMSO, specifically catering to the MoSDeF ecosystem. GMSO satisfies our needs for generality as well as extensibility.
```

Comment:
While the purpose of GMSO is to replace packages like parmed, maybe it would be worth mentioning that it also has the ability to convert to and from parmed?

Reply:
Some additions and edits


Currently, there are data structures designed to represent these systems, such as ParmEd and OpenMM[@shirts2016lessons; @eastmann2017openmm].  However, these data structures are tailored to a specific subset of simulation workflows/ecosystems, and specific force field equation forms, which sacrifice some generality and broad applicability.  This limitation includes, but is not limited to, the hard-coding and assumption of potential (interactional) expressions and units.  They lack the generality that MoSDeF and users need, such as the ability to define and store arbitrary potential expressions or unit systems.  Integrating these new features into existing software would require a major overhaul of these data structures, potentially impacting existing simulation workflows and probably not appealing to current project stakeholders.  Generalizing these potential (force field) expressions also allows users to enter the force field in its native form and units, minimizing user error when setting up the force field file while providing the ability to easily auto-convert the potential form and units to the molecular engine's required form.  Hence, we have developed a new data structure called the General Molecular Simulation Object, or GMSO, specifically catering to the MoSDeF ecosystem.  GMSO satisfies the broader community's need for a general, extensible, and reproducible method of setting up molecular simulations.

Reply:
I added this to a paragraph right after this